be/src/vec/aggregate_functions/aggregate_function_window.h

```diff
@@ -408,61 +489,78 @@ class WindowFunctionData final
+    if (which.idx == TypeIndex::TYPE)                                                       \
+        return new AggregateFunctionTemplate<                                               \
+                Data<LeadAndLagData<TYPE, is_nullable, false, StoreType, init_with_null>>>( \
+                argument_types);
     FOR_NUMERIC_TYPES(DISPATCH)
 #undef DISPATCH
 
     if (which.is_decimal()) {
         return new AggregateFunctionTemplate<
-                Data<LeadAndLagData<Int128, is_nullable, false, StoreType>>>(argument_types);
+                Data<LeadAndLagData<Int128, is_nullable, false, StoreType, init_with_null>>>(
+                argument_types);
     }
     if (which.is_date_or_datetime()) {
         return new AggregateFunctionTemplate<
-                Data<LeadAndLagData<Int64, is_nullable, false, StoreType>>>(argument_types);
+                Data<LeadAndLagData<Int64, is_nullable, false, StoreType, init_with_null>>>(
+                argument_types);
     }
     if (which.is_string_or_fixed_string()) {
         return new AggregateFunctionTemplate<
-                Data<LeadAndLagData<StringRef, is_nullable, true, StoreType>>>(argument_types);
+                Data<LeadAndLagData<StringRef, is_nullable, true, StoreType, init_with_null>>>(
+                argument_types);
     }
     DCHECK(false) << "with unknowed type, failed in  create_aggregate_function_leadlag";
     return nullptr;
 }
 
-template <bool is_nullable, bool is_copy>
+template <bool is_nullable, bool is_copy, bool replace_if_not_null = false>
 AggregateFunctionPtr create_aggregate_function_first(const std::string& name,
                                                      const DataTypes& argument_types,
                                                      const Array& parameters,
                                                      bool result_is_nullable) {
-    return AggregateFunctionPtr(
-            create_function_single_value<WindowFunctionData, WindowFunctionFirstData, is_nullable,
-                                         is_copy>(name, argument_types, parameters));
+    if constexpr (replace_if_not_null) {
+        return AggregateFunctionPtr(
+                create_function_single_value<WindowFunctionData, WindowFunctionFirstNonNullData,
+                                             is_nullable, is_copy, true>(name, argument_types,
+                                                                         parameters));
+    } else {
+        return AggregateFunctionPtr(
+                create_function_single_value<WindowFunctionData, WindowFunctionFirstData,
+                                             is_nullable, is_copy>(name, argument_types,
+                                                                   parameters));
+    }
 }
 
-template <bool is_nullable, bool is_copy>
+template <bool is_nullable, bool is_copy, bool replace_if_not_null = false>
 AggregateFunctionPtr create_aggregate_function_last(const std::string& name,
```

Comment:
maybe should split the register `last_value` and `last_not_null_value`

Reply:
done